File: src/main/kotlin/cz/machovec/lekovyportal/api/service/ProcessedDatasetService.kt
package cz.machovec.lekovyportal.api.service

import cz.machovec.lekovyportal.api.model.ProcessedDatasetResponse
import cz.machovec.lekovyportal.core.domain.dataset.DatasetType
import cz.machovec.lekovyportal.core.repository.ProcessedDatasetRepository
import org.springframework.stereotype.Service

@Service
class ProcessedDatasetService(
    private val repository: ProcessedDatasetRepository
) {
    fun getLatestDataset(types: List<DatasetType>): ProcessedDatasetResponse? {
        return repository.findFirstByDatasetTypeInOrderByCreatedAtDesc(types)
            ?.let {
                ProcessedDatasetResponse(
                    datasetType = it.datasetType,
                    createdAt = it.createdAt,
                    year = it.year,
                    month = it.month
                )
            }
    }
}

File: frontend/src/services/processedDatasetService.ts
// services/processedDatasetService.ts

import { DatasetType } from "../types/DatasetType"

export type ProcessedDatasetResponse = {
    datasetType: DatasetType
    createdAt: string
    year: number
    month: number
}

export async function fetchLatestProcessedDataset(datasetTypes: DatasetType[]): Promise<ProcessedDatasetResponse> {
    const query = datasetTypes.map(t => `types=${t}`).join("&")
    const response = await fetch(`/api/processed-datasets/latest?${query}`)
    if (!response.ok) {
        throw new Error("Failed to fetch latest dataset status")
    }
    return response.json()
}
